Detections/SecurityEvent/password_not_set.yaml

```diff
@@ -25,9 +25,21 @@ relevantTechniques:
 query: |
 
   let starttime = 3d;
-  let SecEvents = materialize ( SecurityEvent | where TimeGenerated >= ago(starttime)
+  let SecEvents = materialize (union isfuzzy=true (
+  SecurityEvent | where TimeGenerated >= ago(starttime)
   | where EventID in (4722,4723) | where TargetUserName !endswith "$"
-  | project TimeGenerated, EventID, Activity, Computer, TargetAccount, TargetSid, SubjectAccount, SubjectUserSid);
+  | project TimeGenerated, EventID, Activity, Computer, TargetAccount, TargetSid, SubjectAccount, SubjectUserSid),
+  ( WindowsEvent | where TimeGenerated >= ago(starttime)
+  | where EventID in (4722,4723) 
+  | extend TargetUserName = tostring(EventData.TargetUserName)
+  | where TargetUserName !endswith "$"
+  | extend Activity=case(EventID==4722,"4722 - A user account was enabled.","4723 - An attempt was made to change an account")
+  | extend TargetAccount = strcat(EventData.TargetDomainName,"\\", EventData.TargetUserName)
+  | extend TargetSid = tostring(EventData.TargetSid)
+  | extend SubjectAccount = strcat(EventData.SubjectDomainName,"\\", EventData.SubjectUserName)
+  | extend SubjectUserSid = tostring(EventData.SubjectUserSid)
+  | project TimeGenerated, EventID, Activity, Computer, TargetAccount, TargetSid, SubjectAccount, SubjectUserSid
+  ));
   let userEnable = SecEvents
```

Comment:
SecEvents does not make sense and is really bad once you add parsing. What happens is that you parse everything twice and then filter half out in each side. The way to solve that is to include your parsre when SecEvent is used, rather than within SecEvent and have two copies.

Reply:
@oshezaf Should I remove materialize all together?